src/control/lib/netdetect/netdetect.go

```diff
@@ -887,37 +993,70 @@ func ScanFabric(provider string) ([]FabricScan, error) {
 	log.Debugf("initDeviceScan completed.  Depth %d, numObj %d, systemDeviceNames %v, hwlocDeviceNames %v",
 		deviceScanCfg.depth, deviceScanCfg.numObj, deviceScanCfg.systemDeviceNames, deviceScanCfg.hwlocDeviceNames)
 
+	var hfiDeviceCount int
+	for _, device := range deviceScanCfg.hwlocDeviceNames {
+		if strings.Contains(device, "hfi1_") {
+			hfiDeviceCount++
+			log.Debugf("There are %d hfi1 devices in the system", hfiDeviceCount)
+		}
+	}
+
```

Comment:
This is reused above, isn't it? Seems also like a good candidate to be shared. Especially since the net result seems to be to output the number of devices found to the log. Should probably move the print outside the loop though.

Reply:
Done.